src/android/FileUtils.java

```diff
@@ -593,14 +593,17 @@ private void getReadPermission(String rawArgs, int action, CallbackContext callb
         if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
             PermissionHelper.requestPermissions(this, requestCode, 
             new String[]{Manifest.permission.READ_MEDIA_IMAGES, Manifest.permission.READ_MEDIA_VIDEO, Manifest.permission.READ_MEDIA_AUDIO});
-          } else {
+        } else {
             PermissionHelper.requestPermission(this, requestCode, Manifest.permission.READ_EXTERNAL_STORAGE);
-          }
+        }
     }
 
     private void getWritePermission(String rawArgs, int action, CallbackContext callbackContext) {
-        int requestCode = pendingRequests.createRequest(rawArgs, action, callbackContext);
-        PermissionHelper.requestPermission(this, requestCode, Manifest.permission.WRITE_EXTERNAL_STORAGE);
+ 	int requestCode = pendingRequests.createRequest(rawArgs, action, callbackContext);
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        } else {
+          PermissionHelper.requestPermission(this, requestCode, Manifest.permission.WRITE_EXTERNAL_STORAGE);
+        }
```

Comment:
I think this can be rewritten as to invert the condition so we don't have an empty code block.

```suggestion
        if (android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
          PermissionHelper.requestPermission(this, requestCode, Manifest.permission.WRITE_EXTERNAL_STORAGE);
        }
```

Reply:
`getWritePermission` is already wrapped by a condition that requires `needPermission` to be `true`.

`needPermission` should then hit this condition:

```
else if(permissionType == WRITE && hasWritePermission()) {
            return false;
        }
```

Which should `return false` once the `hasWritePermission` method is fixed.

I think the only changes needed in this PR should be:

```java
    private boolean hasWritePermission() {
        return android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU
            ? true
            : PermissionHelper.hasPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE);
    }
```

Reply:
@EYALIN Can you review the above messages and make the updates?

I believe you do not need to modify the `getWritePermission` method, as I previously mentioned the reasons why.
Only `hasWritePermission` needs to be updated. The code could be simplified with what I provided in my previous message.

Also, can you rebase your branch against with Cordova's main branch? There are conflicts.

And include Norman's suggestion about including a comment explaining about the `true` return.

```
// Starting with API 33, requesting WRITE_EXTERNAL_STORAGE is an auto permission rejection
```